lib/doc/worksheet.js

```diff
@@ -579,6 +584,29 @@ class Worksheet {
     return image && image.imageId;
   }
 
+  // =========================================================================
+  // Worksheet Protection
+  protect(password, options){
```

Comment:
I know - thanks to Microsoft's "special" implementation of password hashing that it's hard to implement this function in an async way - but I think it would be worth wrapping this function in a Promise.
Calculating the hash takes almost a whole second and this would be a good place to use worker threads if available.